src/config.js

```diff
@@ -18,135 +18,103 @@ class Config {
-    }
-    this.devices[device.deviceName] = device
-    this.save()
-    return log.info('Configuration file successfully updated')
+  // Return the URL of the cozy instance
+  getCozyUrl () {
+    return this.config.url
   }
 
-  // Add remote configuration for a given device name.
-  addRemoteCozy (options) {
-    this.devices[options.deviceName] = options
-    return this.save()
+  // Set the URL of the cozy instance
+  setCozyUrl (url) {
+    this.config.url = url
   }
 
-  // Remove remote configuration for a given device name.
-  removeRemoteCozy (deviceName) {
-    delete this.devices[deviceName]
-    return this.save()
+  // Return true if a device has been configured
+  hasClient () {
+    return !!this.config.client
   }
 
-  // Get Couch URL for given device name.
-  getUrl (deviceName) {
-    if (deviceName == null) { deviceName = this.getDefaultDeviceName() }
-    let device = this.getDevice(deviceName)
-    if (device.url != null) {
-      let url = urlParser.parse(device.url)
-      url.auth = `${deviceName}:${device.passphrase}`
-      return `${urlParser.format(url)}cozy`
-    } else {
-      return null
+  // Return config related to the OAuth client
+  getClient () {
+    if (!this.config.client) {
+      throw new Error(`Device not configured`)
     }
+    return this.config.client
+  }
+
+  // Set the remote configuration
+  saveClient (options) {
+    this.config.client = options
+    this.persist()
   }
 
   // Set the pull, push or full mode for this device
-  // It wan throw an exception if the mode is not compatible with the last
+  // It will throw an exception if the mode is not compatible with the last
   // mode used!
-  setMode (mode, deviceName) {
-    if (deviceName == null) { deviceName = this.getDefaultDeviceName() }
-    if (deviceName && this.devices[deviceName]) {
-      let old = this.devices[deviceName].mode
-      switch (false) {
-        case old !== mode:
-          return true
-        case (old == null):
-          throw new Error('Incompatible mode')
-        default:
-          this.devices[deviceName].mode = mode
-          return this.save()
-      }
-    } else {
-      return false
+  saveMode (mode) {
+    let old = this.config.mode
+    if (old === mode) {
+      return true
+    } else if (old) {
+      throw new Error('Incompatible mode')
     }
+    this.config.mode = mode
+    this.persist()
   }
 
-  // Set insecure flag, for self-signed certificate mainly
-  setInsecure (bool, deviceName) {
-    if (deviceName == null) { deviceName = this.getDefaultDeviceName() }
-    if (deviceName && __guard__(this.devices[deviceName], x => x.url)) {
-      this.devices[deviceName].insecure = bool
-      return this.save()
-    } else {
-      return false
-    }
+  ///////////////////////////////////////////////////////////
+  // Implement the Storage interface for cozy-client-js oauth
+
+  save (key, value) {
+    this.config.credentials[key] = value
+    return Promise.resolve(value)
```

Comment:
Have some kind `Config#oauthStorage()` return the expected interface? (not sure how you expect to use it)